pkg/plugins/runtime/gateway/route_table_generator.go

```diff
@@ -123,7 +124,23 @@ func GenerateVirtualHost(
 			for _, h := range e.RequestHeaders.Replace {
 				switch h.Key {
 				case ":authority", "Host", "host":
-					routeBuilder.Configure(route.RouteReplaceHostHeader(h.Value))
+					// Only one HostRewriteSpecifier can be set for RouteAction
+					// and when rewrite_host_to_backend_hostname is set on
+					// route, it should have precedence over explicit request
+					// host header modification.
+					// It's just precaution as MeshGatewayRoute validator
+					// shouldn't allow above situation to occur.
+					if !e.RewriteHostToBackendHostname {
```

Comment:
Is this required as you do the validation before? Sounds like you probably don't need a WARNING

Reply:
removed